src/executePrettier/executePrettierOnBufferRange.js

```diff
@@ -12,60 +11,80 @@ const buildPrettierStylelintOptions = require('./buildPrettierStylelintOptions')
 const buildPrettierOptions = require('./buildPrettierOptions');
 const handleError = require('./handleError');
 
-const executePrettier = (editor: TextEditor, text: string) =>
-  getPrettierInstance(editor).format(text, buildPrettierOptions(editor));
+const executePrettierWithCursor = (
+  editor: TextEditor,
+  text: string,
+  cursorOffset: number,
+): Prettier$CursorResult =>
+  getPrettierInstance(editor).formatWithCursor(text, {
+    ...buildPrettierOptions(editor),
+    cursorOffset,
+  });
 
-const executePrettierEslint = (editor: TextEditor, text: string) =>
+const executePrettierEslint = (editor: TextEditor, text: string): string =>
   allowUnsafeNewFunction(() => prettierEslint(buildPrettierEslintOptions(editor, text)));
 
-const executePrettierStylelint = (editor: TextEditor, text: string) =>
+const executePrettierStylelint = (editor: TextEditor, text: string): string =>
   prettierStylelint.format(buildPrettierStylelintOptions(editor, text));
 
-const executePrettierOrIntegration = async (editor: TextEditor, text: string) => {
-  try {
-    let formatted;
-    if (shouldUseStylelint() && isCurrentScopeCssScope(editor)) {
-      formatted = await executePrettierStylelint(editor, text);
-    } else if (shouldUseEslint()) {
-      formatted = executePrettierEslint(editor, text);
-    } else {
-      formatted = executePrettier(editor, text);
-    }
+const executePrettierOrIntegration = async (editor: TextEditor, text: string, cursorOffset: number) => {
+  const nextCursorOffset: number = cursorOffset;
+  let nextText;
 
-    return formatted;
-  } catch (error) {
-    return error;
+  if (shouldUseStylelint() && isCurrentScopeCssScope(editor)) {
+    // TODO: add support for cursor position - https://github.com/hugomrdias/prettier-stylelint/issues/13
+    nextText = await executePrettierStylelint(editor, text);
+  } else if (shouldUseEslint()) {
+    // TODO: add support for cursor position - https://github.com/prettier/prettier-eslint/issues/164
+    nextText = executePrettierEslint(editor, text);
+  } else {
+    return executePrettierWithCursor(editor, text, cursorOffset);
   }
+
+  return { formatted: nextText, cursorOffset: nextCursorOffset };
```

Comment:
Let's switch to using returns + guard clauses, I'm not a big fan of `let` although I know this was here before:

```js
const executePrettierOrIntegration = async (editor: TextEditor, text: string, cursorOffset: number) => {
  if (shouldUseStylelint() && isCurrentScopeCssScope(editor)) {
    // TODO: add support for cursor position - https://github.com/hugomrdias/prettier-stylelint/issues/13
    const formatted = await executePrettierStylelint(editor, text);
    return { formatted, cursorOffset };
  } 
  
  if (shouldUseEslint()) {
    // TODO: add support for cursor position - https://github.com/prettier/prettier-eslint/issues/164
    formatted = executePrettierEslint(editor, text);
    return { formatted, cursorOffset };
  } 
  
  return executePrettierWithCursor(editor, text, cursorOffset);
};
```

Reply:
I like this pattern more anyway! I was just trying to adapt based on what was there. Was already having enough fun trying to get flow types figured out that I hadn't taken the time to rethink this here. 👍